sql/core/src/main/scala/org/apache/spark/sql/sources/commands.scala

```diff
@@ -139,9 +140,8 @@ private[sql] case class InsertIntoHadoopFsRelation(
     }
 
     def writeRows(taskContext: TaskContext, iterator: Iterator[Row]): Unit = {
-      writerContainer.executorSideSetup(taskContext)
-
       try {
+        writerContainer.executorSideSetup(taskContext)
```

Comment:
This call also initializes a user defined `OutputWriter`. If an exception is thrown from here, we should fall back to `abortTask()`.